packages/application/test/layoutrestorer.spec.ts

```diff
@@ -60,13 +60,17 @@ describe('apputils', () => {
             collapsed: true,
             currentWidget: null,
             widgets: null,
-            visible: false
+            visible: false,
+            sizes: null,
+            expansionStates: [true, true, true, true]
```

Comment:
Here is the trouble, a sidebar can have multiple panels (called _widgets_ in this object). So `sizes` and `expansionStates` should exist per panel (child of the stack panel).

The panel are identify by a _id_ for example

```json
"left":{
  "collapsed":false,
  "visible":true,
  "current":"filebrowser",
  "widgets":["filebrowser","running-sessions","@jupyterlab/toc:plugin","extensionmanager.main-view"]
},
"right":{
  "collapsed":true,
  "visible":true,
  "widgets":["jp-property-inspector","debugger-sidebar"]
}
```

So I would use a object like

```js
left/rightArea: {
  collapsed: boolean,
  currentWidget: string | null
  widgets: string[] | null,
  widgetStates: {
    [id: string]: {
     sizes: number[] | null,
     expansionStates: boolean[] | null
    }
  }
}
```

Does it make sense?

Reply:
Oh okay. I assumed that each panel would require its own sidebar instance so we would have multiple left and right ones, but this makes more sense. Thanks @fcollonval! I will make the changes.